### PR TITLE
[SPARK-20665][SQL][FOLLOW-UP]Move test case to MathExpressionsSuite

### PR DESCRIPTION
**Pull request by 10110346**
## What changes were proposed in this pull request?

 add test case to MathExpressionsSuite as #17906

## How was this patch tested?

unit test cases

### CODE REVIEW

File: sql/core/src/test/resources/sql-tests/inputs/mathFunctions.sql [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+select ceil(1234567890123456);
+select ceil(12345678901234567);
+select ceiling(1234567890123456);
+select ceiling(12345678901234567);
+
+-- floor
+select floor(0);
+select floor(1);
+select floor(1234567890123456);
+select floor(12345678901234567);
```
Comment (gatorsmile) [inline sql/core/src/test/resources/sql-tests/inputs/mathFunctions.sql]: add a new line here

File: sql/core/src/test/resources/sql-tests/inputs/mathFunctions.sql [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+select bround(3.1415925, 7);
+select bround(3.1415925, 8);
+select bround(3.1415925, 9);
+select bround(3.1415925, 10);
+select bround(3.1415925, 100);
+select bround(3.1415925, 6);
+select bround(null, 8);
+
+-- math functions
```
Comment (gatorsmile) [inline sql/core/src/test/resources/sql-tests/inputs/mathFunctions.sql]: To `-- cot`

**Comment by 10110346** [comment]
Done. Should i delete the unit test case from `MathFunctionsSuite.scala`? 
@gatorsmile

**Comment by rxin** [comment]
Hm I'm not sure if it is a good idea to run so many "unit test" style tests for expressions in the end to end suites. It takes a lot of time than just running unit tests.

**Comment by gatorsmile** [comment]
@10110346 Could you move them to `MathExpressionsSuite`, based on @rxin 's comments? Thanks!

**Comment by 10110346** [comment]
@gatorsmile OK,i will do,thanks

**Comment by 10110346** [comment]
@gatorsmile  Please review it again,thanks

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MathExpressionsSuite.scala [hunk 3]
```diff
@@ -20,9 +20,9 @@ package org.apache.spark.sql.catalyst.expressions
 import java.nio.charset.StandardCharsets
 
 import com.google.common.math.LongMath
-
```
Comment (gatorsmile) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MathExpressionsSuite.scala]: revert this back

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MathExpressionsSuite.scala [hunk 4]
```diff
@@ -250,6 +258,11 @@ class MathExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkConsistencyBetweenInterpretedAndCodegen(Cbrt, DoubleType)
   }
 
+  def checkDataTypeAndCast(expression: UnaryMathExpression): Expression = {
+      val expNew = implicitCast(expression.child, expression.inputTypes(0)).getOrElse(expression)
```
Comment (gatorsmile) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MathExpressionsSuite.scala]: indent issues.

**Comment by gatorsmile** [comment]
ok to test

**Comment by 10110346** [comment]
Test passed, thanks.  @gatorsmile

File: sql/core/src/test/resources/sql-tests/inputs/operators.sql [hunk 5]
```diff
@@ -54,19 +54,4 @@ explain select 3 + 1 || 'a' || 4 / 2;
...
--- ceil and ceiling
-select ceiling(0);
-select ceiling(1);
-select ceil(1234567890123456);
-select ceiling(1234567890123456);
-
--- floor
-select floor(0);
-select floor(1);
-select floor(1234567890123456);
```
Comment (gatorsmile) [inline sql/core/src/test/resources/sql-tests/inputs/operators.sql]: After rethinking it, maybe we should keep this file untouched in this PR. The reason is we still can check the type promotion logics here. 

Yeah, we  still should add the test cases in `MathExpressionsSuite.scala`

**Comment by 10110346** [comment]
@gatorsmile  ok,i will modify and  resubmit

**Comment by gatorsmile** [comment]
Thanks! Merging to master.

**Pull request closed without merging**